### PR TITLE
The Hierophant

### PR DESCRIPTION
**Pull request by ChangelingRain**
New boss, "inspired by"(taken from) Hyper Light Drifter.
- [x] Ruin for boss spawning(don't fight in in the ruin, or do I guess? If you fight it in the ruin there's a good chance you die horribly)
- [x] Reward tuning so the staff is useful as a weapon/mining tool.
- [x] Better sounds for the boss's attacks, possibly distinct sounds for different attacks?
  - [x] Note that the above, while done, isn't implemented, because the sounds involved don't make sense for the current sprites.
- [x] Tuning so it's at least somewhat difficult to fight but not overly difficult, so it can be killed for a decent weapon against other bosses.
- [x] ~~Actual sprites for the boss itself(instead of literally the boss's sprite from Hyper Light Drifter)~~
  - [x] ~~Actual sprites for the weapon it drops(instead of a placeholder that looks like the aforementioned boss's staff)~~
  - [x] Rip those goals it's bird time

### CODE REVIEW

**Comment by PKPenguin321** [comment]
http://hyperlightdrifter.wikia.com/wiki/The_Hierophant

**Comment by ChangelingRain** [comment]
yeah it's that fuck but slightly more mean and it doesn't have any vulture mobs with it

**Comment by Kromgar** [comment]
Ergo already killed it

**Comment by ExcessiveUseOfCobblestone** [comment]
Testmerge Feedback:

- Was Enjoyable to fight and was an interesting boss approach. We fought IN the ruins and we did fine [2 people and a minebot].
- Would like some cosmetic heirophant items to cosplay as the boss I just slayed :dress: 
- Staff is more of a liability unless you're going solo, as unlike the other weapons the staff can create weird patterns without you doing it. If there was some option [possibly tied to a heirophant cosmetic like the hat so you aren't spammed with buttons just for holding the staff] that lowered damage against miners and minebots, that would be appreciated.
- The teleport feature is also solo-oriented, despite the aesthetic that makes it look like a group teleport [which is what i'd prefer anyways]. Sorta misleading [unless it's a reference to the teleport animation from HLD, I never played the game].

Other than that it's a great feature and I hope you continue working on it!

**Comment by ChangelingRain** [comment]
The staff can now toggle friendly fire, and the teleport will teleport all mobs near you.
Those were good ideas, so I've implemented them.

Clothes not included, because the boss's sprite is absolutely not final and I'd rather not have to resprite the clothes to fit whatever the boss becomes.

**Comment by Repukan** [comment]
So many flashing lights, either my internet is shit or the amount of light worms and patterns caused lag.

All in all, pretty fun to play against. Tried soloing it with a proto and got my ass kicked. 10/10 nerf the epilepsy

**Comment by ChangelingRain** [comment]
Lag may or may not be shitcode, I'm not sure how optimized the code this is using is.

**Comment by ChangelingRain** [comment]
Hierophant will now heal for 25% of its missing health if it times out and resets. Letting it reset is bad and should not be abused, mmkay?

**Comment by Kromgar** [comment]
>not letting boss mobs go onto station

Why do you hate fun?

**Comment by PKPenguin321** [comment]
yeah thats lame

**Comment by ChangelingRain** [comment]
If you can get it on the station it'll still be a huge boss but the crew can just try and ignore it and it'll leave if they manage that for long enough. Keywork 'if'

**Comment by ChangelingRain** [comment]
Fixed TK boss cheese(which actually worked on literally every boss provided you kept it right at the edge of sight range)

**Comment by optimumtact** [comment]
stale?

**Comment by ChangelingRain** [comment]
NOT STALE

**Comment by Kromgar** [comment]
I love the Hierophants Staff of Memes

**Comment by ChangelingRain** [comment]
There is an achievement for killing it but the scores/etc from the testing will not carry over to when it's merged.

**Comment by optimumtact** [comment]
Is this ready yet bloody hell

**Comment by ChangelingRain** [comment]
It's waiting on sprites(possibly from kor) and is ready besides those and I'll get super mad if you close it.

**Comment by optimumtact** [comment]
Reopen when you have the sprites.

**Comment by Kromgar** [comment]
@ChangelingRain Jesus fucking christ. What did you do to it? I took it down a couple of days ago was pretty hard.

Faced it as a changeling with survival pens an upgraded ka and a plasma blaster. Suddenly near its end phase where it starts teleporting around.


BOOOM BOOM BOOOM nonstop its summoning massive aoe circles ontop of me. Considering the arena and the size and how rapidly it was putting things out i died with survival pens and fleshmend on my side.

**Comment by ChangelingRain** [comment]
The arena is basically hardmode for it, given how large the aoe blasts get it's sometimes not possible to dodge if you're cornered; this is totally intended if you want to fight it and die less I suggest luring it out of the arena.

**Comment by ChangelingRain** [comment]
It turns out this pr needs to be open to push updates to it for testmerges, as the version on the server should have `Stolen from Hyper Light Drifter. Commit number 58.` as the hierophant's description(the server had this testmerged after the commit that would have done that) and it doesn't.
RIP orange's tracker, i guess.

**Comment by TechnoAlchemisto** [comment]
The sprites from the game don't look that out of place to be quite honest.

**Comment by ChangelingRain** [comment]
Yeah but it's literally ripped from the game and I don't want to literally just take the sprites from the game.

**Comment by Kromgar** [comment]
Copyright theft is not a good thing

**Comment by ChangelingRain** [comment]
Oh hell fuck it I'll get sprites eventually and I can make it what it's supposed to be then(it's supposed to be an alien death robot with some sort of weapon but this huge fucking bird will have to do)

**Comment by TechnoAlchemisto** [comment]
:tada:

**Comment by Kromgar** [comment]
>Makes an arena too small to fight the hierophant in

Whats the point of an arena if its too small?

Why does it spawn a 3x3 directly on top of the player that has no travel time? 


Its literally unkillable at this point you realize that right?

**Comment by Kromgar** [comment]
IF you could just make the arena entirely empty save for 4 pillars the arena might actually be viable to fight the hierophant in

**Comment by ChangelingRain** [comment]
The thing about the arena is, short of it being gigantic and empty(super boring), any arena a player can make will absolutely be better than the possibility of ever being cornered in this arena.
If you remove the possibility of being cornered, then the arena isn't really an arena, it's just a big open space.
Aaaand I don't really like forcing people to fight it in the arena because A: absolutely not the feel of the game B: >totally removing all possibility of getting it to the station for high tier murder

Or in other words I'm cornered into a 'challenge mode' arena by the nature of the game and my own preferences.

File: code/datums/action.dm [hunk 1]
```diff
@@ -189,6 +189,40 @@
...
+	name = "Toggle Friendly Fire \[ON\]"
+	desc = "Toggles if the staff causes friendly fire."
+	button_icon_state = "vortex_ff_on"
+
+/datum/action/item_action/toggle_unfriendly_fire/Trigger()
+	if(..())
+		UpdateButtonIcon()
+
+/datum/action/item_action/toggle_unfriendly_fire/UpdateButtonIcon()
+	if(istype(target, /obj/item/weapon/hierophant_staff))
```
Comment (optimumtact) [inline code/datums/action.dm]: what the fuck is this, shouldn't this be handled by the staff?

Comment (ChangelingRain) [inline code/datums/action.dm]: Having the staff handle it requires a for loop or using locate so this is ~technically better~ even though it's weird

**Pull request merged**